src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -3365,10 +3365,10 @@ void SmartScript::UpdateTimer(SmartScriptHolder& e, uint32 const diff)
             }
         }
 
-        // Delay flee for assist event if stunned or rooted
+        // Delay flee for assist event if stunned, rooted or casting
         if (e.GetActionType() == SMART_ACTION_FLEE_FOR_ASSIST)
         {
-            if (me && me->HasUnitState(UNIT_STATE_ROOT | UNIT_STATE_LOST_CONTROL))
+            if (me && me->HasUnitState(UNIT_STATE_ROOT | UNIT_STATE_LOST_CONTROL | UNIT_STATE_CASTING))
             {
                 e.timer = 1;
```

Comment:
To test this properly you should have a few spell cast SAI actions with small intervals so that they keep failing due to the creature having UNIT_STATE_CASTING.

Reply:
@jackpoz Just letting you know I've seen this response and will try it out.

Reply:
@SaxxonPike

Reply:
@SaxxonPike ? if no answer before 1 april i will close as abandoned.

Reply:
Wow, this one slipped. Will update this week.

Reply:
^^ xd

Reply:
> Wow, this one slipped. Will update this week.

^^